app/views/exercises/show.html.erb

```diff
@@ -4,7 +4,7 @@
 
 <% @stats = @exercise.guide.stats_for(current_user) if current_user && @exercise.guide %>
 
-<% if @exercise.navigable_parent.timed? %>
+<% if @exercise.navigable_parent.timed? && !current_user.teacher?(@exercise.guide.slug) %>
```

Comment:
Perhaps you could rename this method `timed?` to `timed_for?` and pass the user as argument. That way you won't repeat logic